Equilobe.DailyReport.SL/DataService.cs

```diff
@@ -482,7 +482,70 @@ public void SaveIpnLogError(long id, string error)
             }
         }
 
+        public List<SourceControlOptions> GetAllReposSourceControlOptions(long instanceId)
+        {
+            var sourceControlOptions = new List<SourceControlOptions>();
+
+            using (var db = new ReportsDb())
+            {
+                var policyStrings = GetInstancePolicyStrings(instanceId);
+
+                if (!policyStrings.Any())
+                    return sourceControlOptions;
+
+                foreach (var policyString in policyStrings)
+                {
+                    var advancedSettings = new AdvancedReportSettings();
+
+                    Deserialization.XmlDeserialize<AdvancedReportSettings>(policyString)
+                        .CopyPropertiesOnObjects(advancedSettings);
+
+                    sourceControlOptions.Add(advancedSettings.SourceControlOptions);
+                }
+            }
+
+            return sourceControlOptions;
+        }
+
+        public List<User> GetInstanceUsers(long instanceId)
+        {
+            var users = new List<User>();
+
+            using (var db = new ReportsDb())
+            {
+                var policyStrings = GetInstancePolicyStrings(instanceId);
+
+                if (!policyStrings.Any())
+                    return users;
+
+                foreach (var policyString in policyStrings)
+                {
+                    var advancedSettings = new AdvancedReportSettings();
+
+                    Deserialization.XmlDeserialize<AdvancedReportSettings>(policyString)
+                        .CopyPropertiesOnObjects(advancedSettings);
+
+                    users.AddRange(advancedSettings.UserOptions);
+                }
+            }
+
+            return users;
+        }
+
         #region helpers
+        private List<string> GetInstancePolicyStrings(long instanceId)
+        {
+            using (var db = new ReportsDb())
+            {
+                return db.BasicSettings
+                    .Include(p => p.SerializedAdvancedSettings)
```

Comment:
I do not think we need the Include here, right? This is needed only if you close the connection. While you have the connection open, you can access the virtual properties. And you return the strings from here anyway, so you will not access virtual properties outside the connection.